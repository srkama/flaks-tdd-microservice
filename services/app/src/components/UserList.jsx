import React from 'react'

export default function UserList(props) {
    return (
        <>{
            props.users.map((user) => {
                return (
                    <h4
                        key={user.id}
                        className="box title is-4"
                    >
                        {user.username}
                    </h4>
                )
            })
        }</>
    )
}
