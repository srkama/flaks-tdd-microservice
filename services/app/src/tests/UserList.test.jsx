import React from 'react'
import { shallow } from 'enzyme';
import UserList from '../components/UserList';
import renderer from 'react-test-renderer';

const users = [
    { id: '1', 'username': 'kamal', 'email': 'kamal.s@gc.com' },
    { id: '2', 'username': 'kamal1', 'email': 'kamal1.s@gc.com' }]

test('UserLists renders proprely', () => {
    const wrapper = shallow(<UserList users={users} />);
    const element = wrapper.find('h4');
    expect(element.length).toBe(2);
    expect(element.get(0).props.children).toBe('kamal');
    expect(element.get(1).props.children).toBe('kamal1');
})

test('Users snapshots renders properly', () => {
    const tree = renderer.create(<UserList users={users} />).toJSON();
    expect(tree).toMatchSnapshot();
})
