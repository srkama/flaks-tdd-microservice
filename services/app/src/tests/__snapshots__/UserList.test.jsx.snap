// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users snapshots renders properly 1`] = `
Array [
  <h4
    className="box title is-4"
  >
    kamal
  </h4>,
  <h4
    className="box title is-4"
  >
    kamal1
  </h4>,
]
`;
